Document apt-get workaround

a3nm opened the pull request:
Hi, this is as discussed in issue #1. I also documented the need to install the dependencies of vmdb2, if that's OK.

By the way, one other thing that could maybe be documented is the fact that building the image fails mysteriously ("cannot run apt-key") if the user running the build has another umask that 022 (see [here](https://github.com/larswirzenius/vmdb2/issues/26#issuecomment-339157527)). I don't know whether you think it's a relevant point to document, but it's the main problem I had to debug when building the image. What do you think?

a3nm committed:
document apt-get workaround

Fixes: #1

a3nm committed:
document need to install vmdb2 requirements

a3nm committed:
remove unnecessary links

stapelberg commented on README.md:
```diff
@@ -9,11 +9,35 @@ See https://wiki.debian.org/RaspberryPi3#Preview_image for where to obtain the l
+    `apt-get` invocations with `apt-get -o Acquire::Check-Valid-Until=false`
+- If you want to use the latest versions of each software, you can replace
+    `http://snapshot.debian.org/archive/debian/20171007T213914Z/` in raspi3.yaml
+    with the URL of your favorite Debian mirror. Of course, this means that the
```
s,the URL of your favorite Debian mirror,http://deb.debian.org/debian, please. People who care about mirrors will know how to replace it.

a3nm replied:
OK great idea! fixing

stapelberg commented:
Thanks for the PR!

> By the way, one other thing that could maybe be documented is the fact that building the image fails mysteriously ("cannot run apt-key") if the user running the build has another umask that 022 (see here). I don't know whether you think it's a relevant point to document, but it's the main problem I had to debug when building the image. What do you think?

Yeah, we should definitely recommend `env -i` and `umask` until https://github.com/larswirzenius/vmdb2/issues/26 is fixed upstream. Feel free to put that into this very PR, or submit a different one.

a3nm committed:
specify a mirror URL

a3nm committed:
add umask and env -i workaround for vmdb2#26

a3nm committed:
point out that bugs can cause unbootable images

a3nm committed:
clean the environment but keep important variables

building the image fails otherwise (encoding issues, missing PATH,
etc.); took invocation from https://unix.stackexchange.com/a/49057

a3nm commented:
I implemented your suggestion, thanks! As for the environment cleanup issue, I tried to explain how to do it, but with `env -i` it's a bit ugly because you need to keep some variables. What do you think?

I'm running a build now to check that the proposed invocation works until the end; if it does work I'm fine with merging.

stapelberg commented on README.md:
````diff
@@ -9,12 +9,39 @@ See https://wiki.debian.org/RaspberryPi3#Preview_image for where to obtain the l
+
+```shell
+umask 022
+sudo env -i HOME="$HOME" LC_CTYPE="${LC_ALL:-${LC_CTYPE:-$LANG}}" PATH="$PATH" USER="$USER" \
````
Do you know why $HOME and $USER are required? This strikes me as odd.

We should set `LC_CTYPE=C.UTF-8`.

a3nm replied:
$HOME and $USER were probably not required, I just lost patience after $LT_CTYPE and $PATH so I searched for a solution online. Fixed.

a3nm committed:
simplify environment invocation

https://github.com/Debian/raspi3-image-spec/pull/2#discussion_r146799419

a3nm commented:
I did your suggested change, thanks! The build seems to run, just checking now that it actually completes.

stapelberg commented:
Please let me know once you have a working image and I’ll merge. Thanks again!

a3nm commented:
Image creation completed successfully, so for me this can be merged. Thanks for reviewing and for coding this. :)

-- 
Antoine Amarilli
Sent from a mobile device, please excuse brevity and top-posting.

On October 25, 2017 12:15:23 PM GMT+02:00, Michael Stapelberg <notifications@github.com> wrote:
>Please let me know once you have a working image and I’ll merge. Thanks
>again!
